debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerChangeTablePointer.java

```diff
@@ -47,16 +48,26 @@
     private ResultSetMapper<Object[]> resultSetMapper;
     private final ResultSet resultSet;
     private final int columnDataOffset;
+    private final SourceTimestampMode sourceTimestampMode;
 
-    public SqlServerChangeTablePointer(SqlServerChangeTable changeTable, ResultSet resultSet) {
-        this(changeTable, resultSet, COL_DATA);
+    public SqlServerChangeTablePointer(SqlServerChangeTable changeTable, ResultSet resultSet, SourceTimestampMode sourceTimestampMode) {
+        this(changeTable, resultSet, sourceTimestampMode, COL_DATA);
     }
 
-    public SqlServerChangeTablePointer(SqlServerChangeTable changeTable, ResultSet resultSet, int columnDataOffset) {
+    public SqlServerChangeTablePointer(SqlServerChangeTable changeTable, ResultSet resultSet, SourceTimestampMode sourceTimestampMode, int columnDataOffset) {
```

Comment:
The table pointer is just a DTO, making it dependent on the application configuration is undesired.

Reply:
Then how would it know if the last column of the result set is a metadata column?

Reply:
It should unconditionally return the timestamp column from the result set.

Reply:
How is it possible? Each table could have different number of columns. It could be a fixed value only if placed before the data columns. 

Besides returning the timestampl column from the result set it needs to know whether the last column is a timestamp column so that it excludes it from the resulting data returned by `getData` method if it is.

Reply:
I see. In this case, once this code is moved to the mode implementation, we can state that the method can be only called on the result set from the SQL built in the same mode. Otherwise, the behavior is undefined.

Reply:
I still don't know how to replace this constructor argument with the timestamp's column position. Since the select could return variable count of columns the only way to know the timestamp's column position is to call `getColumnCount()` on the result set. So the pseudo code could look like this:

```
int timestampColumnPosition = resultSet.getMetaData().getColumnCount()
new SqlServerChangeTablePointer(..., resultSet, timestampColumnPosition, ...)
```

It's strange to pass a constructor argument (`timestampColumnPosition`) if it can be derived from another constructor argument (`resultSet`). This would have been acceptable if `SqlServerChangeTablePointer` had been used differently in a different place where the `timestampColumnPosition` calculation logic is different, but this is not the case.

Also passign the timestamp's column position as a constructor argument makes internals of `SqlServerChangeTablePointer` much more complex. The code needs to handle different scenarios that won't happen in real, like the timestamp column being in the middle of the data columns, or it's position conflicts with the constants like `SqlServerChangeTablePointer#COL_COMMIT_LSN`.

Reply:
> It's strange to pass a constructor argument (`timestampColumnPosition`) if it can be derived from another constructor argument (`resultSet`).

I don't think this is weird. The component that builds the SQL can derive the number of the column from the total count because it knows that it's the last column. The component that actually returns the timestamp doesn't know if the result set contains the timestamp or not, so passing or not passing the column index would also work as a flag (is there a timestamp or not).